nanofi/include/core/utlist.h

```diff
@@ -0,0 +1,1073 @@
+/*
```

Comment:
I would prefer to place this file somewhere else, as it's not really part of core, but a 3rd party library we use. Could you add to a folder that implies this?

Reply:
Ah thanks for calling this out. I have a more recent commit with a lot of changes, but didn't move this yet and probably would have forgotten.

Reply:
We have a third party dir in the main directory, which is where this should go.

Reply:
Not yet in nanofi :) 
Guess this library is not going to be used in the C++ part.

Reply:
sorry CMAKE_SOURCE_DIR. It can go there and should. I think I updated the NOTICE and we direct people to the root source thirdparty, so we can probably just use that. No reason to create levels of thirdparty source dirs yet.

Reply:
Okay, fine for me. 👍 Happy to approve as it gets moved.

Reply:
As per the title this is to work out a plan so that it's out in the open on Apache. I won't be merging this.